plugin/class-plugin.php

```diff
@@ -58,6 +63,84 @@ private function register_logic() {
 	private function register_interface() {
 		add_filter( 'woocommerce_get_sections_memberships', [ $this, 'add_section' ] );
 		add_filter( 'woocommerce_get_settings_memberships', [ $this, 'get_settings' ], 10, 2 );
+		add_filter(
+			'woocommerce_admin_settings_sanitize_option_' . self::SETTING_ID,
+			[ $this, 'validate_fields_on_update' ],
+			10,
+			3
+		);
+	}
+
+	/**
+	 * Validation of all the email domains settings before they're saved.
+	 *
+	 * @param string $value     Action param, value.
+	 * @param string $option    Action param, option.
+	 * @param string $raw_value Action param, raw value.
+	 *
+	 * @return null|string The `woocommerce_admin_settings_sanitize_option_[OPTION_NAME]` filter expects $value to be returned
+	 *                     if everything's OK, or if null is returned it skips saving this value.
+	 */
+	public function validate_fields_on_update( $value, $option, $raw_value ) {
+
+		$domains = explode( ',', $value );
+
+		// Nothing to save.
+		if ( empty( $domains) ) {
+			return null;
+		}
+
+		foreach ( $domains as $domain ) {
+
+			if ( ! $this->is_valid_domain_entry( $domain ) ) {
+				\WC_Admin_Settings::add_error(
+					__(
+						'ERROR',
+						'newspack_teams_for_wc_memberships_auto_join_by_email'
+					) .
+					' ' .
+					__(
+						'The value you provided is not a valid entry for an email domain:',
+						'newspack_teams_for_wc_memberships_auto_join_by_email'
+					) .
+					': ' . $domain
```

Comment:
You have two `:`, so the error message reads `ERROR The value you provided is not a valid entry for an email domain::`

Reply:
Ah, indeed! Fixed in b052e0d